jmx/jmx.go

```diff
@@ -383,3 +384,17 @@ func receiveResult(lineC chan []byte, queryErrC chan error, cancelFn context.Can
 		}
 	}
 }
+
+func logAvailableWarnings(channel chan string) {
+	var warn string
+	for {
+		select {
+		case warn = <-channel:
+			{
+				log.Warn(warn)
+			}
+		default:
```

Comment:
some time since i don't use the default clause in the select, is it not going to be executed if `channel` is empty? if so i think the for loop wrapping all this is not looping

Reply:
We call this to clean warnings channel, if there is nothing, we are good to break the loop